### PR TITLE
Add missing Ord and ReserveItems impls

### PR DESCRIPTION
**Pull request by antiguru**
Add missing implementations:
* `ReadSlice`: `Ord+PartialOrd+Eq+PartialEq`
* `OptionRegion`: `ReserveItems<Option<T>>`
* `ResultRegion`: `ReserveItems<Result<T, E>>`

### CODE REVIEW

File: src/impls/slice.rs [hunk 1]
```diff
@@ -177,6 +172,45 @@ impl<C: Region, O: OffsetContainer<C::Index>> ReadSlice<'_, C, O> {
...
+where
+    for<'a> R::ReadItem<'a>: PartialEq,
+{
+    fn eq(&self, other: &Self) -> bool {
+        self.iter().eq(*other)
+    }
+}
+
+impl<R: Region, O: OffsetContainer<R::Index>> Eq for ReadSlice<'_, R, O> where
+    for<'a> R::ReadItem<'a>: PartialEq
```
Comment (frankmcsherry) [inline src/impls/slice.rs:186]: Should this be `: Eq`?

**Review by frankmcsherry — APPROVED** [review]
Seems good with one caveat

File: src/impls/slice.rs (hunk 1 above)
Comment (antiguru) [reply src/impls/slice.rs]: Yup, fixed. Thanks!

**Comment by antiguru** [comment]
I had to bump the minimum Rust version to 1.77 because older versions fail to use the `Ord` and related implementations because they can't understand trait bounds on the GAT correctly.

**Pull request merged**